query_autocomplete/fields.nu

```diff
@@ -0,0 +1,68 @@
+# key_words are n1ql query keywords
+const $key_words = [FROM SELECT WHERE]
+
+# cli_operators are the operators that can be suggested to the user as custom completions
+# the only difference between $operators and ¢cli_operators is the = -> ==. This is because =
+# is reserved in nushell and cannot be used as an argument to FROM
+const $cli_operators = [!= <= >= > < == LIKE]
+
+# fields returns the list of valid next words given a partial query
+export def main [context: string] {
+    let $last = ($context | split row " " | drop | last)
+
+    let $query_words = ($context | split words)
+    let $last_word = ($query_words | last)
+    let $used_keywords = ($query_words | each {|it| if ($it in $key_words) {$it}})
+    let $last_keyword = ($used_keywords | last)
+
+    if ($last_word == $last_keyword) and ($last != *) {
+        match $last_word {
+            FROM => {return (collections | get collection)}
+            SELECT => {
+                let $collection = ($context | split words | $in.1)
+                return (collection_fields $collection | prepend *)
+            }
+            WHERE => {
+                let $collection = ($context | split words | $in.1)
+                return (collection_fields $collection)
+            }
+        }
+    }
+
+    let $after_last_keyword = ($context | split row $last_keyword | last)
+
+    match $last_keyword {
+        FROM => {return [SELECT]}
+        SELECT => {
+            # TO DO - make this work with fields with spaces in
```

Comment:
Is there an issue filed for this?

Reply:
Have done: https://github.com/couchbaselabs/couchbase-shell/issues/590